alertmanager: improve docs for updating tenant configuration (#6139)
grafana/mimir: docs/sources/mimir/references/http-api/index.md

POST /api/v1/alerts
```

> Stores or updates the Alertmanager configuration for the authenticated tenant. The Alertmanager configuration is stored in the configured backend object storage and shared between Alertmanager instances.

Reviewer: > and shared between Alertmanager instances.

That seems like unnecessary addition. I mean, how else can it work in Mimir's horizontally scalable Alertmanager?

Author: that was probably just me but that sentence on its own felt like it's just leaking implementation details without a clear purpose (i.e. so what if the config ends up in object storage). But maybe I was too tired when writing this. I can revert this 
```suggestion
Stores or updates the Alertmanager configuration for the authenticated tenant. The Alertmanager configuration is stored in the configured backend object storage.
```

This endpoint expects the Alertmanager **YAML** configuration in the request body and returns `201` on success.

> The names of the templates in `template_files` must be valid file names and not be relative or absolute paths. For example, both `/templates/my-template.tpl` and `./my-template.tpl` are invalid, whereas `my-template.tpl` is valid.

Reviewer: Technically speaking "valid file name" is also a relative path. Perhaps we can say:

```suggestion
The names of the templates in `template_files` must be valid file names and not contain any path separators. For example, both `/templates/my-template.tpl` and `./my-template.tpl` are invalid, whereas `my-template.tpl` is valid.
```
